packages/smarthr-ui/src/components/Select/Select.tsx

```diff
@@ -149,40 +151,52 @@ const ActualSelect = <T extends string>(
         ref={ref}
         className={selectStyle}
       >
-        {hasBlank && (
-          <option value="">{decorators?.blankLabel?.(BLANK_LABEL) || BLANK_LABEL}</option>
-        )}
-        {options.map((option) => {
-          if ('value' in option) {
-            return (
-              <option {...option} key={option.value}>
-                {option.label}
-              </option>
-            )
-          }
-
-          const { options: groupedOptions, ...optgroup } = option
-
-          return (
-            <optgroup {...optgroup} key={optgroup.label}>
-              {groupedOptions.map((groupedOption) => (
-                <option {...groupedOption} key={groupedOption.value}>
-                  {groupedOption.label}
-                </option>
-              ))}
-            </optgroup>
-          )
-        })}
-        {
-          // Support for not omitting labels in Mobile Safari
-          isMobileSafari && <optgroup className={blankOptGroupStyle} />
-        }
+        <BlankOption hasBlank={hasBlank} decorator={decorators?.blankLabel} />
```

Comment:
```suggestion
        hasBlank && <BlankOption decorator={decorators?.blankLabel} />
```

空の選択肢に依存関係があるような場合に、思ったようにレンダリングされない可能性が出そうと思いました。

空選択肢自体のラベルが変わることはなさそうなので、 `hasBlank &&` の判定だけ React.memo() の外に出す、でどうでしょうか？

Reply:
> 空の選択肢に依存関係があるような場合に、思ったようにレンダリングされない可能性が出そうと思いました。

外に出す事自体は問題ないと思うんですが、空の選択肢に依存関係がある場合、というのがちょっと思いつかず...
フォームの入力状況に応じて空の選択肢の有り無しが変わるような状態って無いとは言いませんがほぼないような...?

あとこのmemo化は渡されたパラメータが前回レンダリング時と変化すれば再実行されるため、hasBlankが可変するようなことがあれば問題なく動作するはずです。
例えば true <-> falseの場合はもちろん undefined <-> false の場合も再実行されます。

Reply:
> 渡されたパラメータが前回レンダリング時と変化すれば再実行されるため、hasBlankが可変するようなことがあれば問題なく動作するはず

なるほど。memo化をしっかり理解できていなかったかもしれません。動作的に懸念なければLGTMです！